pkg/provisioner/crdprovisioner.go

```diff
@@ -126,12 +127,36 @@ func (c *CrdProvisioner) CreateVolume(
 	secrets map[string]string,
 	volumeContentSource *v1alpha1.ContentVolumeSource,
 	accessibilityReq *v1alpha1.TopologyRequirement) (*v1alpha1.AzVolumeStatusParams, error) {
+	maxShares := 1
+	var err error
+	if maxSharesStr, ok := parameters["maxShares"]; ok {
```

Comment:
Use `azureutils.MaxSharesField` constant instead of literal.

AFAIK, map searches are case-sensitive and the parameters block passed to `CreateVolume` has keys in lowercase. This may not work correctly.

Reply:
That was actually why I did not use `azureutils.MaxSharesField`  because parameter handling in `CreateVolume` lowercases all the keys in the map in its `select` block. But wouldn't we, in real life scenario, likely use camelcase in our storage class yaml file?

Reply:
The parameter names in the YAML files should be treated as case-insensitive whatever solution we land on here. A Customer could use `maxshares`, `maxShares`, `MaXsHaReS` or any other permutation of casing and still be correct. We need to perform case-insensitive searches then. I had thought CSI might normalize these for use, but the `select` block in the V1 `CreateVolume` implementation suggests otherwise.

Reply:
Makes sense :)